File: flagship/app/models/alert.rb
class Alert < ApplicationRecord
  self.table_name = "submarines.alerts"

  belongs_to :server, foreign_key: :server_id

  # Scopes
  scope :active, -> { where(status: "active") }
  scope :acknowledged, -> { where(status: "acknowledged") }
  scope :resolved, -> { where(status: "resolved") }
  scope :by_severity, ->(severity) { where(severity: severity) }
  scope :recent, -> { order(created_at: :desc) }

  # Status methods
  def acknowledge!
    update(status: "acknowledged", acknowledged_at: Time.current)
  end

  def resolve!
    update(status: "resolved", resolved_at: Time.current)
  end

  # Severity badge color helpers
  def severity_color
    case severity
    when "critical" then "red"
    when "warning" then "yellow"
    when "info" then "blue"
    else "gray"
    end
  end
end

File: flagship/app/Models/Alert.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Alert extends Model
{
    use HasUuids;

    protected $table = 'admiral.alerts';

    protected $fillable = [
        'server_id',
        'alert_type',
        'severity',
        'message',
        'threshold_value',
        'current_value',
        'metadata',
        'status',
        'acknowledged_at',
        'resolved_at',
    ];

    protected $casts = [
        'threshold_value' => 'decimal:2',
        'current_value' => 'decimal:2',
        'metadata' => 'array',
        'acknowledged_at' => 'datetime',
        'resolved_at' => 'datetime',
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
    ];

    public function server(): BelongsTo
    {
        return $this->belongsTo(Server::class);
    }
}
